packages/python/plotly/plotly/express/_core.py

```diff
@@ -812,6 +812,7 @@ def make_trace_spec(args, constructor, attrs, trace_patch):
             or (
                 args["render_mode"] == "auto"
                 and len(args["data_frame"]) > 1000
+                and args["line_shape"] != "spline"
```

Comment:
```suggestion
                and args.get("line_shape") != "spline"
```
Because not all px functions that have `render_mode` also have `line_shape`.

Reply:
Oh good catch, thanks!